pkg/llm-d-inference-sim/simulator.go

```diff
@@ -555,12 +583,15 @@ func (s *VllmSimulator) createCompletionResponse(isChatCompletion bool, respToke
 // according the value of isChatCompletion
 // respTokens - tokenized content to be sent in the response
 // toolCalls - tool calls to be sent in the response
-// model - model name
+// modelName - display name returned to the client and used in metrics. It is either the first alias
+// from --served-model-name (for a base-model request) or the LoRA adapter name (for a LoRA request).
 // finishReason - a pointer to string that represents finish reason, can be nil, stop, length, or tools
 // usageData - usage (tokens statistics) for this response
 func (s *VllmSimulator) sendResponse(isChatCompletion bool, ctx *fasthttp.RequestCtx, respTokens []string, toolCalls []toolCall,
-	model string, finishReason string, usageData *usage) {
-	resp := s.createCompletionResponse(isChatCompletion, respTokens, toolCalls, model, &finishReason, usageData)
+	modelName string, finishReason string, usageData *usage) {
+	// Decide what model name should appear in the JSON payload.
+	displayedModelName := s.getDisplayedModelName(modelName)
```

Comment:
There is no need to call s.getDisplayedModelName here again, since you passed displayModel to this function in reqProcessingWorker, i.e. modelName is already a display name

Reply:
ty @irar2!